From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Felix Hanau <felix@cloudflare.com>
Date: Thu, 30 Jan 2025 14:50:54 -0500
Subject: Fix macOS build

In V8 13.3, absl/synchronization is used on macOS, which was not reflected in
the Bazel build. Also remove the (unused) absl_optional dependency so we can
drop the binding for it.
In 13.4, absl/synchronization is used across platforms and declared in the
Bazel build, absl_optional has been removed and with the switch to bzlmod absl
and hwy bindings as a whole are no longer needed; this patch can be dropped.

Change-Id: I7c52113596247f8f254ac5f882b41da1ba32e3b1

diff --git a/BUILD.bazel b/BUILD.bazel
index 0b95fa690001e06ef39227c66dbd0866d33b2926..42198db354f84480b3a16ef8df2556bbfb65819a 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -4302,6 +4302,10 @@ cc_library(
         "src/torque/kythe-data.h",
         "src/torque/torque-compiler.h",
     ],
+    deps = select({
+        "@v8//bazel/config:is_macos": ["@com_google_absl//absl/synchronization"],
+        "//conditions:default": [],
+    }),
     copts = ["-Wno-implicit-fallthrough"],
     include_prefix = "third_party/v8",
     includes = ["."],
